src/providers/details/tabs/Hosts/components/utils/onSaveHost.ts

```diff
@@ -0,0 +1,158 @@
+  hostPairs: InventoryHostNetworkTriple[];
+  network: NetworkAdapters;
+  user?: string;
+  passwd?: string;
+};
+
+const processHostSecretPair = async (
+  provider: V1beta1Provider,
+  hostPair: InventoryHostNetworkTriple,
+  ipAddress: string,
+  encodedUser: string | undefined,
+  encodedPassword: string | undefined,
+  encodedProvider: string,
+  encodedIpAddress: string,
+) => {
+  const { host, inventory } = hostPair;
+
+  if (host && getName(host)) {
+    // Host already set, update network in the host and secret
+    const { name: secretName, namespace: secretNamespace } = host?.spec?.secret ?? {};
+
+    const secretData = await k8sGet<IoK8sApiCoreV1Secret>({
+      model: SecretModel,
+      name: secretName,
+      ns: secretNamespace,
+    });
+
+    await patchHost(host, ipAddress);
+    await patchSecret(secretData, encodedIpAddress, encodedUser, encodedPassword);
+
+    return;
+  }
+
+  // Create a new host and secret pair
+  // Create a Secret
+  const secretData = {
+    apiVersion: 'v1',
+    data: {
+      ip: encodedIpAddress,
+      password: encodedPassword,
+      provider: encodedProvider,
+      user: encodedUser,
+    },
+    kind: 'Secret',
+    metadata: {
+      generateName: `${provider?.metadata?.name}-${inventory.id}-`,
+      labels: {
+        createdForResource: inventory.id,
+        createdForResourceType: 'hosts',
+      },
+      namespace: getNamespace(provider),
+    },
+    type: 'Opaque',
+  };
+  const createdSecret = await createHostSecret(secretData as IoK8sApiCoreV1Secret);
+
+  // Create Host
+  const newHostData = {
+    apiVersion: 'forklift.konveyor.io/v1beta1',
+    kind: 'Host',
+    metadata: {
+      name: `${provider?.metadata?.name}-${inventory.id}-config`,
+      namespace: getNamespace(provider),
+      ownerReferences: [
+        {
+          apiVersion: 'forklift.konveyor.io/v1beta1',
+          kind: 'Provider',
+          name: getName(provider),
+          uid: getUID(provider),
+        },
+      ],
+    },
+    spec: {
+      id: inventory.id,
+      ipAddress,
+      provider: {
+        name: getName(provider),
+        namespace: getNamespace(provider),
+      },
+      secret: {
+        name: getName(createdSecret),
+        namespace: getNamespace(createdSecret),
+      },
+    },
+  };
+  const createdHost = await createHost(newHostData as V1beta1Host);
+
+  // Patch Secret owner Ref
+  const ownerRef = {
+    name: getName(createdHost),
+    uid: getUID(createdHost),
+  };
+  await patchHostSecretOwner(createdSecret, ownerRef);
+};
+
+/**
+ * Saves hosts data, including associated secrets, for a given set of host+secret pairs.
+ * If a host already exists for a host pair, the host is updated with the new properties.
+ * Otherwise, a new host and secret are created.
+ */
+export const onSaveHost = async ({
+  hostPairs,
+  network,
+  passwd,
+  provider,
+  user,
+}: OnSaveHostParams) => {
+  const encodedUser = user ? Base64.encode(user) : undefined;
```

Comment:
what does Base64.encode(undefined) return ?

Reply:
it returns undefined but eslint doesn't like it.

Reply:
What does the linter complain about?

Reply:
Fixed by replacing with `x ? & decode(x)`